### PR TITLE
add integration test for upgrading from minikube v1.0.0 to latest

### PR DESCRIPTION
**Pull request by medyagh**
to make sure we don't break user configs from v1.0.0 and user be upgradable

### CODE REVIEW

File: test/integration/version_upgrade_test.go [hunk 1]
```diff
@@ -136,3 +140,82 @@ func TestVersionUpgrade(t *testing.T) {
...
+	r := func() error {
+		rr, err = Run(t, exec.CommandContext(ctx, tf.Name(), args...))
+		return err
+	}
+
+	// Retry to allow flakiness for the previous release
+	if err := retry.Expo(r, 13*time.Second, Minutes(30), 2); err != nil {
+		t.Fatalf("v1.0.0 start failed: %v", err)
+	}
+	rr, err = Run(t, exec.CommandContext(ctx, tf.Name(), "stop", "-p", profile))
```
Comment (sharifelgamal) [inline test/integration/version_upgrade_test.go]: Should we check minikube status here for Running as well?

File: test/integration/version_upgrade_test.go [hunk 2]
```diff
@@ -136,3 +140,82 @@ func TestVersionUpgrade(t *testing.T) {
...
+		} `json:"serverVersion"`
+	}{}
+	err = json.Unmarshal(s.Stdout.Bytes(), &cv)
+
+	if err != nil {
+		t.Fatalf("error traversing json output: %v", err)
+	}
+	if _, err = semver.Parse(cv.ServerVersion.GitVersion); err != nil {
+		t.Fatalf("err parsing the kubernets version %v", err)
+	}
```
Comment (sharifelgamal) [inline test/integration/version_upgrade_test.go]: Are we just checking to make sure it's parseable? Should we make sure the k8s version is correct?

Comment (sharifelgamal) [reply test/integration/version_upgrade_test.go]: actually log the k8s version, something like
t.Logf("k8s version is %s", k8sVersion)

File: test/integration/version_upgrade_test.go [hunk 3]
```diff
@@ -136,3 +140,83 @@ func TestVersionUpgrade(t *testing.T) {
...
+		ServerVersion struct {
+			GitVersion string `json:"gitVersion"`
+		} `json:"serverVersion"`
+	}{}
+	err = json.Unmarshal(s.Stdout.Bytes(), &cv)
+	if err != nil {
+		t.Fatalf("error traversing json output: %v", err)
+	}
+	if _, err = semver.Parse(cv.ServerVersion.GitVersion); err != nil {
+		t.Fatalf("err parsing the kubernets version %v", err)
```
Comment (tstromberg) [inline test/integration/version_upgrade_test.go]: Your going to need to strip the leading v:

`version_upgrade_test.go:218: err parsing the kubernets version Invalid character(s) found in major number "v1"`

**Comment by tstromberg** [comment]
Can you please update this test to be shippable? Otherwise we should just close  this.

**Pull request closed without merging**